### DISCUSSION TITLE
Even slower Blackbox logging rate for fixed wing & small dataflash chips

### DISCUSSION THREAD

**Discussion by mirov** [iNavFlight/inav#6776] [Q&A] [answered]
I use a NOXE V1 FC with only 8MB of dataflash for the blackbox logging memory on a fixed wing.  The plane is pretty stable, so I'm not looking to tune PIDs with the data in the logs.  What is useful however is position, altitude, battery, current, etc for long-range (and long duration) flights.  The configuration menu allows 1/32 as the slowest rate, at my loop frequency (1K) that equates to about 31 samples/second.   I could live with a small fraction of that.

Apparently on Betaflight you can set rate_denom with the CLI, but "set rate_denom = 200" for example is rejected (invalid name), even though "get rate_denom" returns a value (though it isn't the 32 I'd expect / hope for).

What's involved in slowing things down ?  Does it also impact consumers of the log data (like mwptools) ?

-Russ

**Comment by JulioCesarMatias** [comment]
hello, I don't understand if you want to increase or decrease the recording speed of BlackBox, but the recording speed will never be greater than 1khz.See this PR for you to understand the reason >> https://github.com/iNavFlight/inav/pull/6127

**Comment by mirov** [comment]
I'd like to keep my loop rate at 1KHz, but throw away most of the data and
log a small fraction of it.  Today we can log *as little* as 1/32 of the
data.  I'd like to (for example) only log 1/200th of the data for a sample
rate of about 5Hz.   Nothing fancy like re-sampling the data to a lower
rate, I'm not worried about aliasing or anything like that.

-Russ

**Comment by JulioCesarMatias** [comment]
use this >> set blackbox_rate_denom = 200

**Comment by mirov** [comment]
Wonderful, thank you.  I guess I was confused and was trying to adjust the
loop rate by mistake.

Thanks again for setting me straight !

-Russ

**Comment by JulioCesarMatias** [comment, marked as answer]
if resolved, put as "Answered"

**Reply by mirov** [reply]
Actually, although the decimation seems to be working I have now down three bench runs of a full battery pack with the plane strapped down for between 90 and 140 min each.  For some reason the BB logs show up as only 70 min each time (give or take a minute).  In each case I ran the battery pack down to well under 10V, but the logs seem to stop recording when they are at about 13V (which is consistent with 70 min into the simulated flight).

So, is there something that might cause this?  The files aren't that big, the 140 min one is only 4.5MB.

**Reply by mirov** [reply]
Hmmm, that limit is suspiciously close to  2^32 microseconds....   I assume that time is being kept in microseconds in a 32 bit value someplace.  Is that someplace inside INAV, or is it an artifact of the BB explorer ?

My use case is to log the telemetry on long (like 1-2 hours) fixed wing flights.

**Reply by mirov** [reply]
For closure, this seems to be a limitation in the v3.4.0 version of Blackbox Explorer.

**Reply by JulioCesarMatias** [reply]
good conclusion, this should be resolved in the future by devs
